subprojects/wrapper/src/main/java/org/gradle/wrapper/Install.java

```diff
@@ -210,30 +208,34 @@ private boolean deleteDir(File dir) {
         return dir.delete();
     }
 
-    private void unzip(File zip, File dest) throws IOException {
-        Enumeration entries;
-        ZipFile zipFile = new ZipFile(zip);
-
+    private void unzip(File zip, File dest) {
         try {
-            entries = zipFile.entries();
-
-            while (entries.hasMoreElements()) {
-                ZipEntry entry = (ZipEntry) entries.nextElement();
-
-                if (entry.isDirectory()) {
-                    (new File(dest, entry.getName())).mkdirs();
-                    continue;
-                }
-
-                OutputStream outputStream = new BufferedOutputStream(new FileOutputStream(new File(dest, entry.getName())));
-                try {
-                    copyInputStream(zipFile.getInputStream(entry), outputStream);
-                } finally {
-                    outputStream.close();
+            Enumeration entries;
+            ZipFile zipFile = new ZipFile(zip);
+            try {
+                entries = zipFile.entries();
+
+                while (entries.hasMoreElements()) {
+                    ZipEntry entry = (ZipEntry) entries.nextElement();
+
+                    if (entry.isDirectory()) {
+                        (new File(dest, entry.getName())).mkdirs();
+                        continue;
+                    }
+
+                    OutputStream outputStream = new BufferedOutputStream(new FileOutputStream(new File(dest, entry.getName())));
+                    try {
+                        copyInputStream(zipFile.getInputStream(entry), outputStream);
+                    } finally {
+                        outputStream.close();
+                    }
                 }
+            } finally {
+                zipFile.close();
             }
-        } finally {
-            zipFile.close();
+        } catch (IOException e) {
+            logger.log("Could not unzip " + zip.getAbsolutePath() + " to " + dest.getAbsolutePath() + ".");
```

Comment:
:x: Shouldn't the unzip operation actually fail the build? Without the unzipped Wrapper we don't have a runtime to execute the build. So I guess in this case we should probably re-throw the exception.

From the test cases it seems to be the case. I just couldn't see how you indicate that to the Gradle process in your changes. Can you point me to it?

Reply:
The exception is now explicitly re-thrown